model/subscription.go

```diff
@@ -62,3 +73,60 @@ func (s *Subscription) Copy(dst datastore.Entity) (datastore.Entity, error) {
 func (s *Subscription) GetCache() datastore.Cache {
 	return nil
 }
+
+// GetSubscription gets a subscription for a given subscriberID (sid) and feedID (fid).
+func GetSubscription(ctx context.Context, store datastore.Store, sid, fid int64) (*Subscription, error) {
+	q := store.NewQuery(typeSubscription, false, "SubscriptionID", "FeedID")
+	q.FilterField("SubscriberID", "=", sid)
+	q.FilterField("FeedID", "=", fid)
+
+	var subscriptions []Subscription
+	_, err := store.GetAll(ctx, q, &subscriptions)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to get subscription with subscriberID: %d, feedID: %d: %w", sid, fid, err)
+	}
+
+	if len(subscriptions) > 1 {
+		return nil, fmt.Errorf("for SubscriberID: %d, and FeedID: %d, failed with error: %w", sid, fid, errDuplicateSubscriptions)
+	}
+
+	return &subscriptions[0], nil
+}
+
+// CreateSubscription creates a subscription for a given subscriber ID and feed ID.
+func CreateSubscription(ctx context.Context, store datastore.Store, sid, fid int64, class, prefs string, renew bool) error {
+	// Calculate characteristics of the subscription.
+	start := time.Now().Truncate(time.Hour * 24) // Start the subscription at the start of the current day.
+	var end time.Time
+	switch class {
+	case SubscriptionDay:
+		end = start.AddDate(0, 0, 1)
+	case SubscriptionMonth:
+		end = start.AddDate(0, 1, 0)
```

Comment:
Worth noting that this normalises the date, so it won't always be the same day in the following month. I don't think that's an issue.